bootstrap/bootstrap.go

```diff
@@ -124,9 +135,24 @@ func (b *Bootstrap) StartEventIngestion(ctx context.Context) error {
 		b.logger,
 	)
 
+	tracer, err := tracers.DefaultDirectory.New(
+		callTracerName,
+		&tracers.Context{},
+		json.RawMessage(callTracerConfig),
+	)
+	if err != nil {
+		return err
+	}
+	blocksProvider := pebble.NewBlocksProvider(
+		b.storages.Blocks,
+		b.config.FlowNetworkID,
+		tracer,
+	)
```

Comment:
can we read the latest block and initialize the blocks provider with it?

Reply:
See https://github.com/onflow/flow-evm-gateway/pull/632#discussion_r1823233170.
The `latestBlock` field of `BlocksProvider` is not the latest indexed block on the EVM Gateway's DB.
It is the latest block the event ingestion engine has received, and the one that should be replayed. If no errors or state mismatches occur, this block will be saved in the EVM Gateway's DB.